structopt-derive/src/attrs.rs

```diff
@@ -291,6 +296,31 @@ impl Attrs {
 
                 VerbatimDocComment(ident) => self.verbatim_doc_comment = Some(ident),
 
+                DefaultValue(ident, lit) => {
+                    let val = if let Some(lit) = lit {
+                        quote!(#lit)
+                    } else {
+                        let ty = if let Some(ty) = self.ty.as_ref() {
+                            ty
+                        } else {
+                            abort!(
+                                ident.span(),
+                                "#[structopt(default_value)] (without an argument) can be used \
+                                only on field level";
+
+                                note = "see \
+                                    https://docs.rs/structopt/0.3.5/structopt/#magical-methods")
+                        };
+                        quote_spanned!(ident.span()=> {
+                            let val = <#ty as ::std::default::Default>::default();
+                            let s = ::std::string::ToString::to_string(&val);
+                            ::std::boxed::Box::leak(s.into_boxed_str())
```

Comment:
What about using a lazy static to have only one allocation by structure (instead of one by call)?

Reply:
I had been thinking about `lazy_static` approach too but I decided against it because I believe that the cons overweight the pros.

Pros:
* Only one leak per `default_value`, usage in loops is safe.

Cons:
* We would need to reexport `lazy_static`, one more dependency, hurts the build time. (Admittedly, not a big concern). 
* This would greatly complicate the code. Current `structopt`'s implementation is nothing but depth-first graph traversing (think about immutable visitors). This structure doesn't play well with popping some info up to the top level; we need all the info on the top because the `lazy_static! { ... }` invocation must be generated just once. 

  I'm not saying this isn't doable, I'm saying this would require a full-scale refactoring. I have one on mind already, but I'd rather postpone it until the release of `clap_derive`, we've been delaying it long enough :)

  Also, this particular problem can be very easily fixed by changing `Arg::default_value` signature to take `Cow<'b str, String>` instead of `'b str`. I believe we should discuss this with the clap-rs team.

Also, I doubt people will complain about this leaks much; even if they will, we can always rework it to using `lazy_static`.

Reply:
You can just use a lazy static exactly here without any refactor:

https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=546d96056c7b65b9eb5198793a40c1d7